include/yaclib/coroutine/detail/future_awaiter.hpp

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include <yaclib/async/future.hpp>
+#include <yaclib/coroutine/detail/coroutine.hpp>
+#include <yaclib/util/detail/atomic_counter.hpp>
+#include <yaclib/util/type_traits.hpp>
+
+#include <iostream>
+
+namespace yaclib::detail {
+
+class FutureAwaiter final {
+ public:
+  FutureAwaiter() = delete;
```

Comment:
You should write assert or this statement.
It is enough to write `static_assert` or `delete` this constructor. Doing both at the same time is pointless.